test/common/testlib.py

```diff
@@ -1146,218 +1175,77 @@ def wrapper(*args):
+        return result
 
+    def run(self, testable):
         tests = []
 
         # The things to test
         def collapse(test, tests):
-            if test.countTestCases() == 1:
+            if isinstance(test, unittest.TestCase):
                 tests.append(test)
             else:
                 for t in test:
                     collapse(t, tests)
         collapse(testable, tests)
-
-        # Now setup the count we have
-        count = len(tests)
-        for i, test in enumerate(tests):
-            setattr(test, "tapOffset", i)
+        test_count = len(tests)
 
         # For statistics
         start = time.time()
-
-        pids = {}
-        options = 0
-        buffer = None
-        if not self.thorough and self.verbosity <= 1:
-            buffer = OutputBuffer()
-            options = os.WNOHANG
-        failures = {"count": 0}
-
-        def join_some(n):
-            while len(pids) > n:
-                if buffer:
-                    buffer.drain()
-                try:
-                    (pid, code) = os.waitpid(-1, options)
-                except KeyboardInterrupt:
-                    sys.exit(255)
-                if code & 0xff:
-                    failed = 1
-                else:
-                    failed = (code >> 8) & 0xff
-                if pid:
-                    if buffer:
-                        output = buffer.pop(pid)
-                        test = pids[pid]
-                        failed, retry = self.filterOutput(test, failed, output)
-                        if retry:
-                            tests.append(test)
-                    del pids[pid]
-                failures["count"] += failed
-
-        while True:
-            join_some(self.jobs - 1)
-
-            if not tests:
-                join_some(0)
-
-                # See if we inserted more tests
-                if not tests:
-                    break
-
+        failures = 0
+        skips = []
+        while tests:
             # The next test to test
-            test = tests.pop()
-
-            # Fork off a child process for each test
-            if buffer:
-                (rfd, wfd) = os.pipe()
-
-            sys.stdout.flush()
-            sys.stderr.flush()
-            pid = os.fork()
-            if not pid:
-                if buffer:
-                    os.dup2(wfd, 1)
-                    os.dup2(wfd, 2)
-                random.seed()
-                offset = getattr(test, "tapOffset", 0)
-                if self.runOne(test, offset):
-                    sys.exit(0)
-                else:
-                    sys.exit(1)
-
-            # The parent process
-            pids[pid] = test
-            if buffer:
-                os.close(wfd)
-                buffer.push(pid, rfd)
+            test = tests.pop(0)
+            result = self.runOne(test)
+            if not result.wasSuccessful():
+                failures += 1
```

Comment:
So "skip" counts as "successful"?

Reply:
Well skip will return 77, if the amount of tests was 1 (so only the skipped test). If there are multiple tests the results of the other tests dictate the value returned